operators/pkg/controller/elasticsearch/client/v6.go

```diff
@@ -29,18 +34,50 @@ func (c *clientV6) UpdateSettings(ctx context.Context, settings Settings) error
 }
 
 func (c *clientV6) ExcludeFromShardAllocation(ctx context.Context, nodes string) error {
-	allocationSetting := ClusterRoutingAllocation{AllocationSettings{ExcludeName: nodes, Enable: "all"}}
-	return c.put(ctx, "/_cluster/settings", allocationSetting, nil)
+	allocationSettings := ClusterRoutingAllocation{
+		Transient: AllocationSettings{
+			Cluster: ClusterRoutingSettings{
+				Routing: RoutingSettings{
+					Allocation: RoutingAllocationSettings{
+						Exclude: AllocationExclude{
+							Name: nodes,
+						},
+					},
+				},
+			},
+		},
+	}
+	return c.put(ctx, "/_cluster/settings", allocationSettings, nil)
 }
 
 func (c *clientV6) EnableShardAllocation(ctx context.Context) error {
-	allocationSetting := ClusterRoutingAllocation{AllocationSettings{Enable: "all"}}
-	return c.put(ctx, "/_cluster/settings", allocationSetting, nil)
-}
-
-func (c *clientV6) DisableShardAllocation(ctx context.Context) error {
-	allocationSetting := ClusterRoutingAllocation{AllocationSettings{Enable: "none"}}
-	return c.put(ctx, "/_cluster/settings", allocationSetting, nil)
+	allocationSettings := ClusterRoutingAllocation{
+		Transient: AllocationSettings{
+			Cluster: ClusterRoutingSettings{
+				Routing: RoutingSettings{
+					Allocation: RoutingAllocationSettings{
+						Enable: "all",
+					},
+				},
+			},
+		},
+	}
+	return c.put(ctx, "/_cluster/settings", allocationSettings, nil)
+}
+
+func (c *clientV6) DisableReplicaShardsAllocation(ctx context.Context) error {
+	allocationSettings := ClusterRoutingAllocation{
+		Transient: AllocationSettings{
+			Cluster: ClusterRoutingSettings{
+				Routing: RoutingSettings{
+					Allocation: RoutingAllocationSettings{
+						Enable: "primaries",
+					},
+				},
+			},
+		},
+	}
+	return c.put(ctx, "/_cluster/settings", allocationSettings, nil)
```

Comment:
Code is factorizable between `EnableShardAllocation` and `DisableReplicaShardsAllocation`.

Reply:
Yes! I'll merge that one and fix it in https://github.com/elastic/cloud-on-k8s/pull/1262.